tutorials/publish_subscribe/java/DeviceStatusSubscriber.java

```diff
@@ -0,0 +1,149 @@
+ *  warranty of any type, including any warranty for fitness for any purpose.
+ *  RTI is under no obligation to maintain or support the Software.  RTI shall
+ *  not be liable for any incidental or consequential damages arising out of the
+ *  use or inability to use the software.
+ */
+
+import java.util.Objects;
+import java.lang.reflect.*;
+
+import com.rti.dds.domain.DomainParticipant;
+import com.rti.dds.domain.DomainParticipantFactory;
+import com.rti.dds.infrastructure.ConditionSeq;
+import com.rti.dds.infrastructure.Duration_t;
+import com.rti.dds.infrastructure.RETCODE_NO_DATA;
+import com.rti.dds.infrastructure.RETCODE_TIMEOUT;
+import com.rti.dds.infrastructure.ResourceLimitsQosPolicy;
+import com.rti.dds.infrastructure.StatusKind;
+import com.rti.dds.infrastructure.WaitSet;
+import com.rti.dds.subscription.InstanceStateKind;
+import com.rti.dds.subscription.ReadCondition;
+import com.rti.dds.subscription.SampleInfo;
+import com.rti.dds.subscription.SampleInfoSeq;
+import com.rti.dds.subscription.SampleStateKind;
+import com.rti.dds.subscription.Subscriber;
+import com.rti.dds.subscription.ViewStateKind;
+import com.rti.dds.topic.Topic;
+
+
+public class DeviceStatusSubscriber extends Application implements AutoCloseable {
+
+    private DomainParticipant participant = null;
+    private DeviceStatusDataReader reader = null;
+    private final DeviceStatusSeq dataSeq = new DeviceStatusSeq();
+    private final SampleInfoSeq infoSeq = new SampleInfoSeq();
+
+    private int processData() {
+        int samplesRead = 0;
+
+        try {
+            reader.take(dataSeq,
+                    infoSeq,
+                    ResourceLimitsQosPolicy.LENGTH_UNLIMITED,
+                    SampleStateKind.ANY_SAMPLE_STATE,
+                    ViewStateKind.ANY_VIEW_STATE,
+                    InstanceStateKind.ANY_INSTANCE_STATE);
+
+            for (int i = 0; i < dataSeq.size(); ++i) {
+                SampleInfo info = infoSeq.get(i);
+
+                if (info.valid_data) {
+                    DeviceStatus deviceStatus = dataSeq.get(i);
+                    if (deviceStatus.is_open) {
+                        System.out.println("WARNING: " + deviceStatus.sensor_name
+                                + " in " + deviceStatus.room_name + " is open!");
+                    }
+
+                }
+                samplesRead++;
+            }
+        } catch (RETCODE_NO_DATA noData) {
+            // No data to process, not a problem
+        } finally {
+            reader.return_loan(dataSeq, infoSeq);
+        }
+
+        return samplesRead;
+    }
+
+    private void runApplication() {
+        participant = Objects.requireNonNull(
+            DomainParticipantFactory.get_instance().create_participant(
+                0,
+                DomainParticipantFactory.PARTICIPANT_QOS_DEFAULT,
+                null, // listener
+                StatusKind.STATUS_MASK_NONE));
+
+        Subscriber subscriber = Objects.requireNonNull(
```

Comment:
We're not using the subscriber or publisher in the examples unless they're strictly needed. Call create_datareader from the participant.